File: frontend/next-generation-ticketing-system/src/Entity/RouteDistrict.ts
import Route from "./Route";
import District from "./District";

class RouteDistrict {
  routeDistrictId: number;
  route: Route;
  district: District;
  distOrder: number;

  constructor(
    routeDistrictId: number,
    route: Route,
    district: District,
    distOrder: number
  ) {
    this.routeDistrictId = routeDistrictId;
    this.route = route;
    this.district = district;
    this.distOrder = distOrder;
  }
}

export default RouteDistrict;

File: backend/ngts/src/main/java/com/three/ngts/Entity/RouteDistrict.java
package com.three.ngts.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "route_district")
public class RouteDistrict {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "route_district_id")
    private Long routeDistrictId;

    @ManyToOne
    @JoinColumn(name = "route_id")
    private Route route;

    @ManyToOne
    @JoinColumn(name = "dist_id")
    private District district;

    @Column(name = "dist_order")
    private Long distOrder;
}
